mesonbuild/dependencies/misc.py

```diff
@@ -43,7 +43,7 @@ def netcdf_factory(env: 'Environment', for_machine: 'MachineChoice',
     if language not in ('c', 'cpp', 'fortran'):
         raise DependencyException(f'Language {language} is not supported with NetCDF.')
 
-    candidates = []  # type: T.List['DependencyType']
+    candidates: 'TV_DepGenerators' = []
```

Comment:
I find these `TV_*` arguments really not helpful. It's not at all obvious what is allowed here, as opposed to `T.List[T.Callable[[], 'ExternalDependency']` which makes it super obvious what this is a list of. I'd really prefer that an alias have a descriptive name like `T.List[CallbleReturnsDependency]`. These are as much for me the author of the code to understand what's going on as it is for the type checker.

Reply:
I primarily changed it so that the type is defined in one place, and the `TV_` prefix is because it is used in multiple locations already. Though I agree that something like `T.List[DependencyGenerator]` is a lot nicer in this case.

Reply:
Changed it to `T.List['DependencyGenerator']`

Reply:
sounds good, thanks